zookeeper/src/main/java/site/ycsb/db/zookeeper/ZKClient.java

```diff
@@ -0,0 +1,213 @@
+import java.util.Vector;
+import java.util.concurrent.TimeUnit;
+
+import org.apache.zookeeper.CreateMode;
+import org.apache.zookeeper.KeeperException;
+import org.apache.zookeeper.WatchedEvent;
+import org.apache.zookeeper.Watcher;
+import org.apache.zookeeper.ZooDefs;
+import org.apache.zookeeper.ZooKeeper;
+import org.json.simple.JSONObject;
+import org.json.simple.JSONValue;
+
+import site.ycsb.ByteIterator;
+import site.ycsb.DB;
+import site.ycsb.DBException;
+import site.ycsb.Status;
+import site.ycsb.StringByteIterator;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * YCSB binding for <a href="https://zookeeper.apache.org/">ZooKeeper</a>.
+ *
+ * See {@code zookeeper/README.md} for details.
+ */
+public class ZKClient extends DB {
+
+  private ZooKeeper zk;
+  private Watcher watcher;
+
+  public static final String CONNECT_STRING = "zookeeper.connectString";
+  public static final String DEFAULT_CONNECT_STRING = "127.0.0.1:2181";
+  public static final String SESSION_TIMEOUT_PROPERTY = "zookeeper.sessionTimeout";
+  public static final long DEFAULT_SESSION_TIMEOUT = TimeUnit.SECONDS.toMillis(30L);
+  public static final String WATCH_FLAG = "zookeeper.watchFlag";
+
+  public static final Charset UTF_8 = Charset.forName("UTF-8");
+  private static final Logger LOG = LoggerFactory.getLogger(ZKClient.class);
+
+  public void init() throws DBException {
+    Properties props = getProperties();
+
+    String connectString = props.getProperty(CONNECT_STRING);
+    if (connectString == null || connectString.length() == 0) {
+      connectString = DEFAULT_CONNECT_STRING;
+    }
+
+    String watchFlagString = props.getProperty(WATCH_FLAG);
+    if (watchFlagString == null || "false".equals(watchFlagString)) {
+      watcher = null;
+    } else if ("true".equals(watchFlagString)) {
+      watcher = new SimpleWatcher();
+    } else {
+      throw new IllegalArgumentException("zookeeper.watchFlag only has two options:true or false");
+    }
+
+    long sessionTimeout;
+    String sessionTimeoutString = props.getProperty(SESSION_TIMEOUT_PROPERTY);
+    if (sessionTimeoutString != null) {
+      sessionTimeout = Integer.parseInt(sessionTimeoutString);
+    } else {
+      sessionTimeout = DEFAULT_SESSION_TIMEOUT;
+    }
+
+    try {
+      zk = new ZooKeeper(connectString, (int) sessionTimeout, new SimpleWatcher());
+    } catch (IOException e) {
+      throw new DBException("Creating connection failed.");
+    }
+  }
+
+  public void cleanup() throws DBException {
+    try {
+      zk.close();
+    } catch (InterruptedException e) {
+      throw new DBException("Closing connection failed.");
+    }
+  }
+
+  @Override
+  public Status read(String table, String key, Set<String> fields,
+                     Map<String, ByteIterator> result) {
+    String path = getPath(key);
+    try {
+      byte[] data = zk.getData(path, watcher, null);
+      if (data == null || data.length == 0) {
+        return Status.OK;
+      }
```

Comment:
if there's no data for the given key we should return `Status.NOT_FOUND`

Reply:
Done